Sga 21 global leaderboard

GillWell opened the pull request:
![image_2024-04-24_150217790](https://github.com/PVP-K409/fit-flow/assets/126980964/ef491e73-be37-4e2c-8dff-a1121ee449a9)

GillWell committed:
Global leaderboard

Added a function to the UserRepository that gets the list of user document information. Uses that information for the global leaderboard.

Global leaderboard shows every user.

Leaderboard card has shows the placing, user URL, username or email, the amount of xp the user has

GillWell committed:
Global leaderboard

-Moved the leaderboard to the "You" screen in the activities tab
-Renamed the activities tab to the global leaderboard
-Changed the leaderboard to show the top 5 users and then five more users(current user being between two users)
- if the user is in the top 5, then it only shows the top five users
- if he is top 6 it doesn't show 4 and 5 place users again
- if the current user is at the bottom of the leader board it shows the current user and 2 users that have more xp than him

GillWell committed:
🤖 Apply Spotless

GillWell committed:
Update YouScreen.kt

GillWell committed:
Merge branch 'SGA-21-Global-Leaderboard' of https://github.com/PVP-K409/fit-flow into SGA-21-Global-Leaderboard

ugnspa commented:
Maybe align the star icon.

Add some padding to cards.

Also if i am in top 5 i am not available to scroll. I suggest making the "top 5" list at the top and "you placing" where you could scroll 50 user upwards and 50 users downwards (you can choose the number by yourself). For instance if I am 25th i would be able to see from 1st to 75th. and if im 150th i would be able to see from 100th to 200th or something like that.

GillWell committed:
Global leaderboard changes

- Shows 50 more users even if the user is in the top 5
- If the user is not in the top five. It will show the top 5 users then 25 users above and 25 users below the current user

GillWell committed:
🤖 Apply Spotless

mantasjasikenas committed:
Minor styling adjustments to make similar to progress screen

mantasjasikenas committed:
Merge branch 'dev' into SGA-21-Global-Leaderboard

mantasjasikenas committed:
🤖 Apply Spotless
